### PR TITLE
feat(frontend/expr): boolean constant folding for (Is, IsNot) x (True, False, Null)

### PR DESCRIPTION
**Pull request by Enter-tainer**
## What's changed and what's your intention?

- Summarize your change (**mandatory**)

Add constant folding for `IsTrue`, `IsFalse`, `IsNull`, `IsNotTrue`, `IsNotFalse`, `IsNotNull`. `NULL` is taken into account as well. For example, `NULL IS NOT TRUE` will become `True`.

## Checklist

- [x] I have written the necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: src/frontend/src/expr/utils.rs [hunk 1]
```diff
@@ -99,24 +99,72 @@ impl ExprRewriter for BooleanConstantFolding {
...
+                }
+                if let Some(v) = try_get_bool_constant(input) {
+                    return ExprImpl::literal_bool(v);
+                }
+            }
+            Type::IsNotNull => {
+                let input = inputs.first().unwrap();
+                if try_get_bool_constant(input).is_some() {
+                    return ExprImpl::literal_bool(true);
+                }
```
Comment (neverchanje) [inline src/frontend/src/expr/utils.rs:156]:
```suggestion
                if let v = try_get_bool_constant(input) {
                    return ExprImpl::literal_bool(v.is_some());
                }
```

Actually IsNotNull optimization can be applied on any constant folding, not just bool. I think eventually the code should be:

```rust
if let v = fold_constant(input) { // fn fold_constant(input) -> DatumRef
  return ExprImpl::literal_bool(v.is_some());
}
```

Comment (Enter-tainer) [reply src/frontend/src/expr/utils.rs:156]: > Actually IsNotNull optimization can be applied on any folded value, not just bool.

That makes sense. It should work on any non-null constant. I will fix it.

Comment (Enter-tainer) [reply src/frontend/src/expr/utils.rs]: Done in 5278ef0844e2e7822e9d4907daa4b927895e0d42.

`inputs` have been folded before(https://github.com/singularity-data/risingwave/pull/2265/files#diff-ae37135915c851ac6f44591a162bd326715ebbc24f279723c795826ddcae3232R85 FYI) so I only check whether `input` is null or not.

**Review by neverchanje — APPROVED** [review]
lgtm

**Pull request merged**